rewrite-test/src/main/java/org/openrewrite/test/TypeValidation.java

```diff
@@ -49,4 +49,17 @@ public class TypeValidation {
     public static TypeValidation none() {
         return new TypeValidation(false,false,false,false,false,false);
     }
+
+    static TypeValidation before(RecipeSpec testMethodSpec, RecipeSpec testClassSpec) {
+        TypeValidation typeValidation = testMethodSpec.getBeforeTypeValidation() != null ?
+                testMethodSpec.getBeforeTypeValidation() : testClassSpec.getBeforeTypeValidation();
+        // TODO: Default to `after()` is for backwards compatibility
+        return typeValidation != null ? typeValidation : after(testMethodSpec, testClassSpec);
```

Comment:
Having the `before` validations default to the `after` validations is probably just confusing. It does have the advantage of offering backwards compatibility, however.

I am thinking we might want to integrate it like this first to make sure all receipe repos build successfully and then make the two validations independent in a second step. But unsure about that as well.